SYCL fallback of volatile_load doesn't work with arbitrary types.

The changes in https://github.com/kokkos/kokkos/pull/9219 don't work well with Kokkos Kernels.
https://my.cdash.org/builds/3691789/build shows
```
Build succeeded.
In file included from /.../benchmark_testing/kokkos-kernels_build/sparse/eti/generated_specializations_cpp/gauss_seidel_numeric/Sparse_gauss_seidel_numeric_eti_DOUBLE_ORDINAL_INT_OFFSET_INT_LAYOUTLEFT_EXECSPACE_SYCL_MEMSPACE_SYCLSPACE.cpp:8:
In file included from /.../benchmark_testing/sparse/impl/KokkosSparse_gauss_seidel_spec.hpp:8:
In file included from /.../benchmark_testing/kokkos-install/include/Kokkos_Core.hpp:32:
In file included from /.../benchmark_testing/kokkos-install/include/KokkosCore_Config_DeclareBackend.hpp:7:
In file included from /.../benchmark_testing/kokkos-install/include/decl/Kokkos_Declare_SERIAL.hpp:8:
In file included from /.../benchmark_testing/kokkos-install/include/Serial/Kokkos_Serial.hpp:28:
In file included from /.../benchmark_testing/kokkos-install/include/impl/Kokkos_HostThreadTeam.hpp:9:
In file included from /.../benchmark_testing/kokkos-install/include/Kokkos_Atomic.hpp:36:
In file included from /.../benchmark_testing/kokkos-install/include/Kokkos_Atomics_Desul_Wrapper.hpp:15:
/.../benchmark_testing/kokkos-install/include/impl/Kokkos_Volatile_Load.hpp:32:5: error: no matching constructor for initialization of 'Kokkos::pair<int, int>'
   32 |   T old = *src_ptr;
      |     ^     ~~~~~~~~
```

--- a/core/src/impl/Kokkos_Volatile_Load.hpp
+++ b/core/src/impl/Kokkos_Volatile_Load.hpp
@@ -29,7 +29,7 @@ namespace Kokkos {
     KOKKOS_COMPILER_INTEL_LLVM < 20260000
 template <typename T>
 KOKKOS_FORCEINLINE_FUNCTION T volatile_load(T const volatile* const src_ptr) {
-  T old = *src_ptr;
+  T old = const_cast<const T&>(*src_ptr);
   T assumed;
   do {
     assumed = old;

